tests/utils/dev-server.js

```diff
@@ -63,6 +66,9 @@ const startServer = async ({ cwd, env = {}, args = [] }) => {
     })
     ps.catch((error) => !selfKilled && reject(error))
   })
+
+  const { timeout, ...server } = await Promise.race([serverPromise, getTimeoutPromise(SERVER_START_TIMEOUT)])
```

Comment:
[sand] Do you think `p-timeout` might help with this pattern?
https://github.com/sindresorhus/p-timeout

Reply:
Done in https://github.com/netlify/cli/pull/1694/commits/12b4af197cbfc48ec946ccbf15dc05da571f8aa9